runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolBase.java

```diff
@@ -134,6 +134,15 @@ public static ResponsePayloadMsg getVersionResponseMsg(VersionInfo vi) {
                 .build();
     }
 
+    /**
+     * Returns a Java VersionInfo object from its Protobuf representation.
+     *
+     * @param msg   the Protobuf VERSION response message
+     */
+    public static VersionInfo getVersionInfo(VersionResponseMsg msg) {
+        final Gson parser = new GsonBuilder().create();
+        return parser.fromJson(msg.getJsonPayloadMsg(), VersionInfo.class);
```

Comment:
won't this throw a parser exception if the json is malformed?

Reply:
Done. Added a try-catch block.